###### QTM 385

***

## Homework 1

Student: [2302860]


***

### Question 1

Run the following code:

```
0.1 + 0.1 + 0.1
```

Is it equal to `0.3`? If not, why?

In [1]:
# Your answer here
print(0.1 + 0.1 + 0.1)
print('0.1 + 0.1 + 0.1 = 0.3 is '+str((0.1 + 0.1 + 0.1) == 0.3)+'\n')

0.30000000000000004
0.1 + 0.1 + 0.1 = 0.3 is False



We think of `0.1` as a rational number, a ratio of two number 1 and 10. We have no issue as we typically think with a base of 10. However, computers do not use a base 10 system; instead, they use a base 2 system (binary code). Computers cannot exactly have the number `0.1` as `0.1` cannot be mathematically expressed with the base 2 system. Thus, adding `0.1` three times will give a result close to `0.3` but not exactly `0.3`.


***

### Question 2

How many digits has $3^{1,000,000}$? (Hint: the function `str` transforms a number into a string.)

In [2]:
# Your answer here
len(str(3**1000000))

477122

$3^{1,000,000}$ has a total of `477,122` digits


***

### Question 3

Create a tuple with three foods that you love to eat. Save it in a variable called `preferred_food`.

In [3]:
# Your answer here
preferred_food = ("apple", "orange", "apricot")
print('preferred_food = '+str(preferred_food)+'\n')

preferred_food = ('apple', 'orange', 'apricot')




***

### Question 4

The module random has a handy function for selecting elements in a tuple: `choice`. Import the module `random`, and name it `rd`.

In [4]:
# Your answer here
import random as rd
print('importing module `random` as rd')

importing module `random` as rd



***

### Question 5

Using the function choice, select a food to eat today. (Hint: when we used `numpy`, I named it `np`. When I needed a function in it, I did `np.function_I_needed`...)

In [5]:
# Your answer here
rd.choice(preferred_food)
print('I will eat an '+str(rd.choice(preferred_food))+' today \n')

I will eat an apple today 



**That's all, folks!**